lib/src/lib.rs

```diff
@@ -761,7 +770,8 @@ mod test {
         let mut initiator =
             EdhocInitiator::new(state, I, G_R, ID_CRED_I, CRED_I, ID_CRED_R, CRED_R);
 
-        let message_1 = initiator.prepare_message_1();
+        let c_i: u8 = generate_connection_identifier_cbor().into();
```

Comment:
Is `.into()` needed here (and other similar lines)? The output of `generate_connection_identifier_cbor()` is already a `u8`.

Reply:
Whether the output of generate_connection_identifier is U8 or u8 used to depend on the back-end; those can go away when rebasing onto remove-hacspec, which is my next step.

Reply:
Oh indeed, great!